Assets/Script/Navigator/GuideBotController.cs

```diff
@@ -0,0 +1,136 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+using UnityEngine.Events;
+
+namespace Neogoma.Stardust.Demo.Navigator
+{
+    public class GuideBotController : MonoBehaviour
+    {
+        //List of waypoints to go through.
+        public List<Vector3> waypointList = new List<Vector3>();
+        //moving speed of  guide bot
+        public float moveSpeed;
+        //rotation speed of guide bot
+        public float rotSpeed;
+        //distance from next target to assume guide bot is close enough.
+        public float distanceFromTarget;
+        //defines if guide bot can move to next point.
+        public bool canMove;
+        //defines if guide bot can look at player camera.
+        public bool canLook;
+
+        private bool isReached1stNode;
+        private Vector3 closestTarget;
+        private int currIndex = 0;
+        private float step;
+
+        public delegate void OnFinishedNav( );
+        
+        public static event OnFinishedNav OnFinishedNavigation;
+        private void Start()
+        {
+            step = moveSpeed * Time.deltaTime;
+        }
+
+        void Update()
+        {
+            if (canLook)
+            {
+                RotateToTarget(Camera.main.transform.position);
+            }
+
+            if (canMove == true)
```

Comment:
Once again, this can be written in this way

```suggestion
            if (canMove)
```